beater/config/auth.go

```diff
@@ -90,14 +89,16 @@ type AnonymousAgentAuth struct {
 	AllowService []string  `config:"allow_service"`
 	RateLimit    RateLimit `config:"rate_limit"`
 
-	configured bool // anon explicitly defined
+	enabledSet bool // enabled explicitly set.
+	configured bool // auth.anonymous is explicitly set.
```

Comment:
I had to keep the configured bool here to allow us to log deprecation logs and let users know when their deprecated settings are ignored when auth.anonymous is explicitly configured.